fix(compiler-cli): adding references to const enums in runtime code

crisbeto opened the pull request:
We had a couple of places where we were assuming that if a particular symbol has a value, then it will exist at runtime. This is true in most cases, but it breaks down for `const` enums.

Fixes #38513.

JoostK commented on packages/compiler-cli/test/ngtsc/ngtsc_spec.ts:
```diff
@@ -4369,6 +4369,54 @@ runInEachFileSystem(os => {
          expect(jsContents).toMatch(setClassMetadataRegExp('type: undefined'));
        });
 
+    it('should use `undefined` in setClassMetadata if enums can\'t be represented as values',
```
nit: I'd personally mention "const enums" here explicitly.

```suggestion
    it('should use `undefined` in setClassMetadata for const enums',
```

JoostK reviewed (approved):
Minor nit, otherwise LGTM

crisbeto committed:
fix(compiler-cli): adding references to const enums in runtime code

We had a couple of places where we were assuming that if a particular
symbol has a value, then it will exist at runtime. This is true in most cases,
but it breaks down for `const` enums.

Fixes #38513.
